droidlet/memory/craftassist/swarm_worker_memory.py

```diff
@@ -0,0 +1,186 @@
+import pdb, sys
+from multiprocessing import Queue
+from droidlet.shared_data_structs import Time
+from typing import Optional, List, Tuple, Sequence, Union
+from droidlet.base_util import XYZ, Block, npy_to_blocks_list
+import pickle
+import uuid
+from droidlet.memory.craftassist.mc_memory_nodes import (  # noqa
+    # DanceNode,
+    VoxelObjectNode,
+    # BlockObjectNode,
+    # BlockTypeNode,
+    # MobNode,
+    # ItemStackNode,
+    # MobTypeNode,
+    # InstSegNode,
+    # SchematicNode,
+    # NODELIST,
+)
+
+NONPICKLE_ATTRS = [
+    "agent",
+    "memory",
+    "agent_memory",
+    "tasks_fn",
+    "run_condition",
+    "init_condition",
+    "remove_condition",
+    "stop_condition",
+    "movement",
+]
+
+class ForkedPdb(pdb.Pdb):
+    """A Pdb subclass that may be used
+    from a forked multiprocessing child
+    """
+
+    def interaction(self, *args, **kwargs):
+        _stdin = sys.stdin
+        try:
+            sys.stdin = open("/dev/stdin")
+            pdb.Pdb.interaction(self, *args, **kwargs)
+        finally:
+            sys.stdin = _stdin
+
+class SwarmWorkerMemory():
+    """Represents the memory for the agent in Minecraft"""
+
+    def __init__(self,
+        memory_send_queue,
+        memory_receive_queue,
+        # db_file=":memory:",
+        # db_log_path=None,
+        # schema_paths=SCHEMAS,
+        # load_minecraft_specs=True,
+        # load_block_types=True,
+        # preception_range=PERCEPTION_RANGE,
+        agent_time=None,
+        # coordinate_transforms=None,
+        # agent_low_level_data={},
+    ):
+        self.send_queue = memory_send_queue
+        self.receive_queue = memory_receive_queue
+        self.receive_dict = {}
+        self.init_time_interface(agent_time)
+        self._safe_pickle_saved_attrs = {}
+        mem_id_len = len(uuid.uuid4().hex)
+        self.self_memid = "0" * (mem_id_len // 2) + uuid.uuid4().hex[: mem_id_len - mem_id_len // 2]
```

Comment:
should we use a different self_memid for swarm workers?

Reply:
I'm using different id because It creates PlayerNode when instantiating the worker. Using the same id would cause database error.